pkg/codec/encodings/array.go

```diff
@@ -0,0 +1,60 @@
+package encodings
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+)
+
+func NewArray(numElements int, underlying TypeCodec) (TypeCodec, error) {
+	if underlying == nil {
+		return nil, fmt.Errorf("%w: field type cannot be nil", types.ErrInvalidConfig)
+	}
+
+	return &array{
+		NumElements: numElements,
+		Field:       underlying,
+	}, nil
+}
+
+type array struct {
+	NumElements int
+	Field       TypeCodec
+}
+
+var _ TypeCodec = &array{}
+
+func (a *array) Encode(value any, into []byte) ([]byte, error) {
+	rValue := reflect.ValueOf(value)
+	if rValue.Kind() != reflect.Array && rValue.Kind() != reflect.Slice {
+		return nil, types.ErrNotASlice
```

Comment:
How about including the kind in the error?
```suggestion
	if kind := rValue.Kind(); kind != reflect.Array && kind != reflect.Slice {
		return nil, fmt.Errorf("%w: expected array or slice but got %s", types.ErrNotASlice, kind)
```